src/main/scala/com/databricks/labs/mosaic/core/geometry/MosaicGeometryESRI.scala

```diff
@@ -160,7 +160,9 @@ abstract class MosaicGeometryESRI(geom: OGCGeometry) extends MosaicGeometry {
             OperatorFactoryLocal.getInstance.getOperator(Operator.Type.Intersection).asInstanceOf[OperatorIntersection]
         val cursor: GeometryCursor =
             op.execute(geom.getEsriGeometryCursor, another.getEsriGeometryCursor, geom.getEsriSpatialReference, null, -1)
-        OGCGeometry.createFromEsriCursor(cursor, geom.getEsriSpatialReference, true)
+        val result = OGCGeometry.createFromEsriCursor(cursor, geom.getEsriSpatialReference, true)
+        if (result.isEmpty) { this.getGeom.intersection(another) }
```

Comment:
Is the logic:
1. intersect geom with another (line 162)
2. result assigned from cursor: null means you reached the end,  empty means no intersection (line 163)
3. if  result is empty, recurse and intersect next_geom (from cursor) with another (line 164)
 
I am thinking specifically of a geometry collection. Is it that a geometry collection is flattened ( from PR #330 ) before this operation is called, so really only testing 1 geometry ever? If the collection is not flattened, why wouldn't the behavior be to accumulate all non-empty results over a geometry collection and then return the union of all as the the intersection answer?

Reply:
The issue is that sometimes the operator returns empty geom even though the intersection is not empty. 
This is a workaround the issue. It isnt an ideal fix. But I dont see a more stable fix given what I know about the dependency.